### PR TITLE
core: add trace_utils module

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
Add a trace_utils module that includes and will include helpers for
writing integrations.

In this patch helpers for calculating a service name are included.

### CODE REVIEW

File: ddtrace/contrib/trace_utils.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+"""
+This module contains utility functions for writing ddtrace integrations.
```
Comment (jd) [inline ddtrace/contrib/trace_utils.py:2]: `trace_utils` vs `utils`?
I mean it's `ddtrace`, not `ddcookpotatoes` so we know it's it's utils lib for tracing ;)

**Review by jd — COMMENTED** [review]
it'd be cool to see how it's used at the same time in existing code, no?

File: ddtrace/contrib/trace_utils.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply ddtrace/contrib/trace_utils.py:2]: :shrug: we have a track record of having `util` or `utils` all over the repo, thought it wouldn't hurt to be a bit more verbose.

**Comment by Kyle-Verhoog** [comment]
@jd see #1612 for usage in existing code 🙂

**Review by jd — APPROVED** [review]
Yeah I saw it after, fine with me :)

**Pull request merged**